queries/yaml/injections.scm

```diff
@@ -25,3 +25,14 @@
              (block_sequence_item
                (block_node
                   (block_scalar) @bash (#offset! @bash 0 1 0 0))))))
+
+;; Prometheus Alertmanager ("expr")
+(block_mapping_pair
+  key: (flow_node) @_expr (#eq? @_expr "expr")
+  value: (flow_node
+           (plain_scalar) @promql))
```

Comment:
I am a bit hesitant on this capture. Shouldn't this be more narrowed down as there should be lots of cases where the block in question here isn't promql?

Reply:
It could match non-promql stuff i think; but wouldnt same apply to the `bash` captures at the top of the file?

Reply:
Well yes, but bash is more used than promql (I believe).

I'm just a bit against it because I have no idea how often this is used, and how often it will lead to unwanted promql injections. For `bash` captures at least there's a ton of workflow files on Github and GitLab to check for usage to check for frequency

Reply:
Would a more specific match be acceptable?

Reply:
Anything that narrows it down to be more specific will be preferred.

If the current queries are of such state, I'd like to have some guidelines and documentations for promql usage that uses such captures.

Reply:
Ok, ill narrow it down some more and write a unittest! Sorry i didn't get the last part, can you rephrase please?

Reply:
What I mean is that if you believe that the current version of the queries are suffice, I'd like to see some documentations and guidelines on why that's the case. I'm okay with the queries right now, though that's with some concerns on the frequency of mis-injection

Reply:
Oh , I mainly just tweaked until highlighting looked fine, it was not a rigorous process. Or do you mean specifically for the injections?

Reply:
Yes, I mean the injections. Is there any documentations or guidelines on the possible key-value pairs that should have its value be `promql` code, and if so, what are the chances of such key-value pairs not being promql code.

If the false injection rate is too high, I'd like to drop the injections, and you can add it locally in your config. The highlights looks fine to me

Reply:
The injections are intended for https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/ ( and alerting rules );

Reply:
Lgtm.

Reply:
I didnt manage to write a more scoped injection and couldnt get dynamic detection with the "@language" directive to work. Should i take the injections out and define them locally?

Reply:
if in doubt, be conservative, especially with injections (which can be performance killers). it's always possible to add them later (which can then be in the upstream format, saving me from having to rewrite them for the `main` branch).

Reply:
Oh, sorry i didnt get the last part; whats upstream format? I dont feel too bad about the injections so far; if they are contentious i can take them out and define them locally but since there is no "promql" filetype; the only place where i would want the parser to be active is injected into yaml right now.

Reply:
I mean the format described in the [official tree-sitter docs](https://tree-sitter.github.io/tree-sitter/syntax-highlighting#language-injection), which we're switching to as soon as we drop support for Neovim 0.8.x.

And no filetype is not a blocker for adding the parser; you can always register it with `vim.filetype.add` (or `vim.treesitter.start` manually), but if injection is the _only_ real use case for this parser, leaving them out is indeed somewhat pointless.

Reply:
Ah, what i mean is that those promql statements do not appear as a format of a whole file; they only ever appear in those yaml injections ( to my knowledge at least )